examples/wasm-example/src/services.rs

```diff
@@ -71,3 +80,105 @@ pub(crate) async fn subscribe_to_finalized_blocks(
     }
     Ok(())
 }
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(js_name = getAccounts)]
+    pub fn js_get_accounts() -> Promise;
+    #[wasm_bindgen(js_name = signPayload)]
+    pub fn js_sign_payload(payload: String, source: String, address: String) -> Promise;
+}
+
+/// DTO to communicate with JavaScript
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Account {
+    /// account name
+    pub name: String,
+    /// name of the browser extension
+    pub source: String,
+    /// the signature type, e.g. "sr25519" or "ed25519"
+    pub ty: String,
+    /// ss58 formatted address as string. Can be converted into AccountId32 via it's FromStr implementation.
+    pub address: String,
+}
+
+pub async fn get_accounts() -> Result<Vec<Account>, anyhow::Error> {
+    let result = JsFuture::from(js_get_accounts())
+        .await
+        .map_err(|js_err| anyhow!("{js_err:?}"))?;
+    let accounts_str = result
+        .as_string()
+        .ok_or(anyhow!("Error converting JsValue into String"))?;
+    let accounts: Vec<Account> = serde_json::from_str(&accounts_str)?;
+    Ok(accounts)
+}
+
+fn to_hex(bytes: impl AsRef<[u8]>) -> String {
+    format!("0x{}", hex::encode(bytes.as_ref()))
+}
+
+fn encode_to_hex<E: Encode>(input: &E) -> String {
+    format!("0x{}", hex::encode(input.encode()))
+}
+
+/// this is used because numeric types (e.g. u32) are encoded as little-endian via scale (e.g. 9430 -> d6240000)
+/// while we need a big-endian representation for the json (e.g. 9430 -> 000024d6).
+fn encode_to_hex_reverse<E: Encode>(input: &E) -> String {
+    let mut bytes = input.encode();
+    bytes.reverse();
+    format!("0x{}", hex::encode(bytes))
+}
```

Comment:
Yeah, it's really because we aren't scale encoding some of these numbers at all; we're just providing the hex representation of them. So really we should just `hex::encode` without the `input.encode()` bit :)

Reply:
Maybe the functions should be:

```rust
fn scale_encoded_hex<E: Encode>(input: &E) -> String { ... }
fn hex_encoded_bytes(input: &[u8]) -> String { ... }
```

And then `.to_be_bytes` or whatever on the numbers we pass to the latter?